src/ScottPlot/Plottable/HelperAlgorithms/CloseToSegmentChecker.cs

```diff
@@ -0,0 +1,106 @@
+        /// <param name="coordinateY">point y coordinate</param>
+        /// <param name="snapX">permissible deviation for x</param>
+        /// <param name="snapY">permissible deviation for y</param>
+        public CloseToSegmentChecker(double coordinateX, double coordinateY, double snapX, double snapY)
+        {
+            CoordinateX = coordinateX;
+            CoordinateY = coordinateY;
+            SnapX = snapX;
+            SnapY = snapY;
+
+            _boundaryPolys = new (double x, double y)[][]
+                {
+                    new (double x, double y)[]
+                    {
+                        (0, 0),
+                        (0, 0),
+                        (0, 0),
+                        (0, 0),
+                    },
+                    new (double x, double y)[]
+                    {
+                        (0, 0),
+                        (0, 0),
+                        (0, 0),
+                        (0, 0),
+                    }
+                };
+        }
+
+        /// <summary>
+        /// Check for point close to segment
+        /// </summary>
+        /// <param name="x">segment first point x coordinate</param>
+        /// <param name="y">segment first point y coordinate</param>
+        /// <param name="x1">segment second point x coordinate</param>
+        /// <param name="y1">segment second point y coordinate</param>
+        /// <returns>true if point close to segment</returns>
+        public bool IsClose(double x, double y, double x1, double y1)
+        {
+            // fix first poly
+            _boundaryPolys[0][0].x = x;
+            _boundaryPolys[0][0].y = y + SnapY;
+
+            _boundaryPolys[0][1].x = x;
+            _boundaryPolys[0][1].y = y - SnapY;
+
+            _boundaryPolys[0][2].x = x1;
+            _boundaryPolys[0][2].y = y1 - SnapY;
+
+            _boundaryPolys[0][3].x = x1;
+            _boundaryPolys[0][3].y = y1 + SnapY;
+
+            // fix second poly
+            _boundaryPolys[1][0].x = x - SnapX;
+            _boundaryPolys[1][0].y = y;
+
+            _boundaryPolys[1][1].x = x + SnapX;
+            _boundaryPolys[1][1].y = y;
+
+            _boundaryPolys[1][2].x = x1 + SnapX;
+            _boundaryPolys[1][2].y = y1;
+
+            _boundaryPolys[1][3].x = x1 - SnapX;
+            _boundaryPolys[1][3].y = y1;
+
+
+            for (int k = 0; k < _boundaryPolys.Length; k++)
+            {
+                bool inside = false;
+                for (int j = 0, j1 = _boundaryPolys[k].Length - 1; j < _boundaryPolys[k].Length; j1 = j++)
+                {
+                    if ((_boundaryPolys[k][j].y > CoordinateY) != (_boundaryPolys[k][j1].y > CoordinateY) &&
+                        (CoordinateX < (_boundaryPolys[k][j1].x - _boundaryPolys[k][j].x) * (CoordinateY - _boundaryPolys[k][j].y) / (_boundaryPolys[k][j1].y - _boundaryPolys[k][j].y) + _boundaryPolys[k][j].x))
+                    {
+                        inside = !inside;
```

Comment:
I may be confused about what this section is doing... but should this be `inside |= inside`?

Reply:
oops, that was a pretty silly comment. I'm going to delete it and let's pretend it didn't happen 😅